core/domain/classifier_services.py

```diff
@@ -417,12 +445,24 @@ def _add_examples(self, examples, iterations):
     def add_examples_for_training(self, training_examples):
         """Adds examples to the classifier with _training_iterations number of
         iterations.
+
+        Args:
+            training_examples: A list of docs.
+
+        Returns:
+            A xrange object. An iterator over the ids of the given docs.
         """
         return self._add_examples(training_examples, self._training_iterations)
 
     def add_examples_for_predicting(self, prediction_examples):
         """Adds examples to the classifier with _prediction_iterations number
         of iterations.
+
+        Args:
+            prediction_examples: A list of docs.
+
+        Returns:
+            A xrange object, the prediction report (iterator of doc ids).
```

Comment:
Now i know it's a **mistake** i misunderstood what prediction report was. This function simply returns an iterator of the added docs. The prediction report is returned by **_get_prediction_report_for_doc**.